src/demux/ts-demuxer.ts

```diff
@@ -859,8 +871,35 @@ class TSDemuxer extends BaseDemuxer {
                     let descriptors = data.subarray(i + 5, i + 5 + ES_info_length);
                     this.dispatchPESPrivateDataDescriptor(elementary_PID, stream_type, descriptors);
                 }
-            } else if (stream_type === StreamType.kID3) {
-                pmt.timed_id3_pids[elementary_PID] = true;
+            } else if (stream_type === StreamType.kMetadata) {
+                if (ES_info_length > 0) {
+                    // parse descriptor for PES private data
+                    for (let offset = i + 5; offset < i + 5 + ES_info_length; ) {
+                        let tag = data[offset + 0];
+                        let length = data[offset + 1];
+
+                        if (tag === 0x26) {
+                            let metadata_application_format = (data[offset + 2] << 8) | (data[offset + 3] << 0);
+                            let metadata_application_format_identifier = null;
+                            if (metadata_application_format === 0xFFFF) {
+                                metadata_application_format_identifier = String.fromCharCode(... Array.from(data.subarray(offset + 4, offset + 4 + 4)));
+                            }
+                            let metadata_format = data[offset + 4 + (metadata_application_format === 0xFFFF ? 4 : 0)];
+                            let metadata_format_identifier = null;
+                            if (metadata_format === 0xFF) {
+                                metadata_format_identifier = String.fromCharCode(... Array.from(data.subarray(offset + 4 + (metadata_application_format === 0xFFFF ? 4 : 0) + 1, offset + 4 + (metadata_application_format === 0xFFFF ? 4 : 0) + 5)));
```

Comment:
This line is too long. You'd better refine it.

Reply:
fixed